File: 804_Unique_Morse_Code_Words/xsy/solution.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        codes = set()
        for word in words:
            # 'a' is 97
            temp = map(lambda i: morse[ord(i)-97], word)
            codes.add("".join(temp))
        return len(codes)

File: 804_Unique_Morse_Code_Words/xsy/solution.go
func uniqueMorseRepresentations(words []string) int {
    morse := []string{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."}
	codes := map[string]int{}
	for _, word := range words {
		temp := ""
		for _, w := range word {
			temp += morse[rune(w)-97]
		}
		codes[temp] += 1
	}
	return len(codes)
}
